### ISSUE TITLE
Return value of ITX\Jobapplications\Utility\GoogleIndexingApiConnector::findBestPluginPageFit() must be an instance of ITX\Jobapplications\Utility\mixed, string returned

### ISSUE THREAD

**Issue by peerlight** [itx-informationssysteme/jobapplications#65]
When extension setting google.indexing_api is activated I get this 503 in backend after saving a posting.

This is an TYPO3 Installation with two sites and therefor two list an detail plugins for jobapplications. Error also arises when one of them is hidden.

PHP 7.4
TYPO3 11.5.10
jobapplications 2.0.0

**Comment by peerlight** [comment]
more detail error log sadly without providing mor info ...
```

Mon, 16 May 2022 10:49:29 +0200 [CRITICAL] request="d7bab94e043b4" component="TYPO3.CMS.Core.Error.ProductionExceptionHandler": Core: Exception handler (WEB: BE): TypeError, code #0, file /var/www/wwwhome/SITE/public_html/cms/typo3conf/ext/jobapplications/Classes/Utility/GoogleIndexingApiConnector.php, line 272: Return value of ITX\Jobapplications\Utility\GoogleIndexingApiConnector::findBestPluginPageFit() must be an instance of ITX\Jobapplications\Utility\mixed, string returned- TypeError: Return value of ITX\Jobapplications\Utility\GoogleIndexingApiConnector::findBestPluginPageFit() must be an instance of ITX\Jobapplications\Utility\mixed, string returned, in file /var/www/wwwhome/SITE/public_html/cms/typo3conf/ext/jobapplications/Classes/Utility/GoogleIndexingApiConnector.php:272 - {"mode":"WEB","application_mode":"BE","exception_class":"TypeError","exception_code":0,"file":"/var/www/wwwhome/SITE/public_html/cms/typo3conf/ext/jobapplications/Classes/Utility/GoogleIndexingApiConnector.php","line":272,"message":"Return value of ITX\\Jobapplications\\Utility\\GoogleIndexingApiConnector::findBestPluginPageFit() must be an instance of ITX\\Jobapplications\\Utility\\mixed, string returned","request_url":"https://SITE/typo3/record/edit?token=--AnonymizedToken--&edit%5Btx_jobapplications_domain_model_posting%5D%5B27%5D=edit&route=%2Frecord%2Fedit&returnUrl=%2Ftypo3%2Fempty%3Ftoken%3D--AnonymizedToken--","exception":"TypeError: Return value of ITX\\Jobapplications\\Utility\\GoogleIndexingApiConnector::findBestPluginPageFit() must be an instance of ITX\\Jobapplications\\Utility\\mixed, string returned in /var/www/wwwhome/SITE/public_html/cms/typo3conf/ext/jobapplications/Classes/Utility/GoogleIndexingApiConnector.php:272\nStack trace:\
n#0 /var/www/wwwhome/SITE/public_html/cms/typo3conf/ext/jobapplications/Classes/Utility/GoogleIndexingApiConnector.php(149): ITX\\Jobapplications\\Utility\\GoogleIndexingApiConnector->findBestPluginPageFit(Array, Object(ITX\\Jobapplications\\Domain\\Model\\Posting))\
n#1 /var/www/wwwhome/SITE/public_html/cms/typo3conf/ext/jobapplications/Classes/Hooks/TCEmainHook.php(105): ITX\\Jobapplications\\Utility\\GoogleIndexingApiConnector->updateGoogleIndex(27, false)\
n#2 /var/www/wwwhome/SITE/public_html/typo3_src-11.5.10/typo3/sysext/core/Classes/DataHandling/DataHandler.php(1094): ITX\\Jobapplications\\Hooks\\TCEmainHook->processDatamap_postProcessFieldArray('update', 'tx_jobapplicati...', 27, Array, Object(TYPO3\\CMS\\Core\\DataHandling\\DataHandler))\
n#3 /var/www/wwwhome/SITE/public_html/typo3_src-11.5.10/typo3/sysext/backend/Classes/Controller/EditDocumentController.php(563): TYPO3\\CMS\\Core\\DataHandling\\DataHandler->process_datamap()\
...
```

**Comment by peerlight** [comment]
return type "mixed" is available from PHP 8.0 onwards; not in PHP 7.4
